### DISCUSSION TITLE
New Mod Proposition: "Bubble Mod"

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#20441] [Ideas] [answered]
Since its already coded ( https://github.com/mk56-spn/osu/tree/Echo-Mod-Implementation for anyone wishing to give it a shot ) ill let the video do the talking:

https://user-images.githubusercontent.com/74463310/198165803-9a38bfba-9948-4a6d-a558-872b2b1904dd.mov

some side notes:

**_what does this mod add to merit its existence?_**

It adds another option to the rather scarce pool of Overlay mods, in other words mods that overlay a series of objects to ( usually ) increase difficulty.

**_What does it do different from flashlight and blinds_**
In flashlight the majority of difficulty comes from placing your cursor correctly on objects you haven't been able to see yet because you have not passed over that part of the playfield since they faded in, in blinds  its well pretty self explanatory.

This kind of takes the opposite sort  of  approach to flashlight where difficulty is concerned , giving a much harder time to players on objects that come after objects in places you have recently tapped.

This post is mainly to gauge interest in this potential addition to the mod roster. and for feedback regarding its functionality, thanks for reading!

**Comment by ghost** [comment]
Are you going to add support for sliderbodies?

My main issue with the demo is how it looks. maybe a better idea to just increase the size of the original hitcircle rather than fading it out.

**Reply by mk56-spn** [reply]
hitcircles can have transparent parts. and older hitcircles dont go over new ones. so thats not viable

**Comment by iSimpFor** [comment]
i think that if you want the mod to do the opposite of flashlight, it would be a good idea to make the echoes fade in from the outside rather than solid

i think it's a good mod that can create a different reading challenge compared to flashlight

**Reply by mk56-spn** [reply]
its not strictly made for that, just a side effect i noticed during making it

**Reply by mk56-spn** [reply]
btw mind making a smal sketch of what your proposed change is exactly, i dont think i quite grasp it

**Reply by iSimpFor** [reply]
![fade](https://user-images.githubusercontent.com/44514667/192115108-1f0216f4-ca90-488a-a7da-69ed71078e4f.png)
something like this

**Reply by mk56-spn** [reply]
hmm personally not a fan, in any case this is different enough that they dont really overlap in functionality  , this suggestion being a lot closer to fl than the current ver

**Comment by rozukke** [comment]
The fact that the blocker circles fall/go only in one direction adds a bit of unnecessary bias to how it plays imo. Doing a quick fade at the end of each one seems like it would make the difficulty a bit more well rounded and wouldn't make the lower half of the playfield comparatively harder.

**Reply by mk56-spn** [reply]
i used to have this, but fading out creates ugly overlaps.but i see your point, maybe i could make it go to the nearest side of the screen to level it out

**Comment by mk56-spn** [comment]
Updated the circles to fade out instead of move out of the playfield as per popular request  and worked on how it looks at different ARs

https://user-images.githubusercontent.com/74463310/193682764-aa5d38c8-893f-41be-9eac-fbdf8bd82181.mp4

any additional feedback would be appreciated

**Reply by ghost** [reply]
are approach circles disabled in the mod or is hidden being used in the videos

**Reply by mk56-spn** [reply]
im using hidden because i expect it to be preferred, im not outright removing them since i can see people wanting to try it with TC or simply use it with approach circles to learn / have a less memory heavy version

**Comment by Walavouchey** [comment]
this looks more like something that should be done through skinning, rather than be its own mod. rn you could implement something similar by skinning the hitbursts

**Reply by mk56-spn** [reply]
I think you're missing the entire point of this. The circles are overlayed and add considerable difficulty. I see no reason this would be in skinning. You could also vaguely achieve what traceable does with a transparent hitcircle yet it's not a skin either because of the difficulty it adds

in addition hitbursts dont last remotely aslong as what is needed for this effect.

**Reply by mk56-spn** [reply]
For reference look at the new vid in the post right above this one. As you can see it's obscuring upwards of 50% of the playfield at times

**Comment by mk56-spn** [comment, marked as answer]
Will be closed by : #21789
